### PR TITLE
feat(fork): Fork monitor

### PR DESCRIPTION
**Pull request by faultytolly**
# PR Standards

## Opening a pull request should be able to meet the following requirements

--

PR naming convention: https://hackmd.io/@nZpxHZ0CT7O5ngTp0TP9mg/HJP_jrm7A

---

Close #XXX

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: block/fork.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	"fmt"
+	"time"
+
+	"github.com/dymensionxyz/dymint/node/events"
+	"github.com/dymensionxyz/dymint/types"
+	uevent "github.com/dymensionxyz/dymint/utils/event"
+)
+
+// MonitorForkUpdate listens to the hub
+func (m *Manager) MonitorForkUpdate(ctx context.Context) error {
```
Comment (srene) [inline block/fork.go:14]: not sure if its done on purpose, but its MonitorForkUpdate not called when starting the manager

File: block/fork.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		m.freezeNode(ctx)
+	}
+
+	return nil
+}
+
+func (m *Manager) createInstruction(rollapp *types.Rollapp, block *types.Block) error {
+	info, err := m.SLClient.GetStateInfo(block.Header.Height)
```
Comment (srene) [inline block/fork.go:60]: if the state info is only necessary for the next proposer, there is already implemented GetNextProposer() func

File: settlement/local/local.go [hunk 3]
```diff
@@ -54,6 +54,16 @@ type Client struct {
 	settlementKV store.KV
 }
 
+func (c *Client) GetStateInfo(index uint64) (*types.StateInfo, error) {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (c *Client) GetRollapp() (*types.Rollapp, error) {
```
Comment (srene) [inline settlement/local/local.go:62]: i guess this will need to be implemented, otherwise it will not be possible to use mock settlement.

**Comment by mtsitrin** [comment]
handled over https://github.com/dymensionxyz/dymint/pull/1199

**Pull request closed without merging**